### PR TITLE
[Merged by Bors] - feat: add perfectly normal (T_6) topological property

### PR DESCRIPTION
**Pull request by StevenClontz**
Introduces the final whole-number separation axiom: [perfectly normal](https://topology.pi-base.org/properties/P000015) / [$T_6$](https://topology.pi-base.org/properties/P000067). Includes a proof that perfectly normal implies completely normal (resp. $T_6$ implies $T_5$). Future work: show all pseudometrizable spaces are perfectly normal (resp. all metrizable spaces are $T_6$).

---

- [x] depends on: #13176
- [x] depends on: #13694

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Separation.lean [hunk 1]
```diff
@@ -2455,6 +2458,104 @@ instance [CompletelyNormalSpace X] [R0Space X] : T5Space (SeparationQuotient X)
...
+    · use fun n ↦ (closure (f' n))ᶜ
+      constructor
+      · rw [← compl_iInter, subset_compl_comm, S_int']
+        exact Disjoint.subset_compl_left (Disjoint.symm hd₁)
+      intro n
+      constructor
+      · simp only [isOpen_compl_iff, isClosed_closure]
+      · simp only [closure_compl, disjoint_compl_left_iff_subset]
+        exact Subset.trans subset_closure
+            (Subset.trans (cls_sub_f' n) (subset_interior_closure (f'_open n)))
```
Comment (StevenClontz) [inline Mathlib/Topology/Separation.lean:2546]: TODO: figure out how to DRY these arguments

**Comment by PatrickMassot** [comment]
bors d+

**Comment by StevenClontz** [comment]
bors merge

**Pull request closed without merging**